lib/mayaUsd/ufe/UsdMaterialHandler.cpp

```diff
@@ -0,0 +1,52 @@
+// ===========================================================================
+// Copyright 2022 Autodesk, Inc. All rights reserved.
+//
+// Use of this software is subject to the terms of the Autodesk license
+// agreement provided at the time of installation or download, or which
+// otherwise accompanies this software in either electronic or hard copy form.
+// ===========================================================================
+
+#include "UsdMaterialHandler.h"
+
+#include <mayaUsd/ufe/UsdSceneItem.h>
+
+#include <pxr/usd/usdGeom/imageable.h>
+
+namespace MAYAUSD_NS_DEF {
+namespace ufe {
+
+UsdMaterialHandler::UsdMaterialHandler()
+    : Ufe::MaterialHandler()
+{
+}
+
+UsdMaterialHandler::~UsdMaterialHandler() { }
+
+/*static*/
+UsdMaterialHandler::Ptr UsdMaterialHandler::create()
+{
+    return std::make_shared<UsdMaterialHandler>();
+}
+
+//------------------------------------------------------------------------------
+// UsdMaterialHandler overrides
+//------------------------------------------------------------------------------
+
+Ufe::Material::Ptr UsdMaterialHandler::material(const Ufe::SceneItem::Ptr& item) const
+{
+    UsdSceneItem::Ptr usdItem = std::dynamic_pointer_cast<UsdSceneItem>(item);
+#if !defined(NDEBUG)
```

Comment:
The directive is useless as the `assert` is gone in release mode. Anyway the code must return if `usdItem` is null.

Reply:
The equivalent in USD land is TF_AXIOM. Grep a bit to see how it is used. Requires the directive to use the PXR namespace.